installing

ubuntu 24.04

go install github.com/pashkov256/deletor@latest
go: downloading github.com/pashkov256/deletor v1.3.0
go: downloading github.com/Bios-Marcel/wastebasket/v2 v2.0.3
go: downloading github.com/schollz/progressbar/v3 v3.14.2
go: downloading github.com/fatih/color v1.16.0
go: downloading github.com/charmbracelet/bubbletea v0.25.0
go: downloading github.com/charmbracelet/lipgloss v0.9.1
go: downloading github.com/charmbracelet/bubbles v0.18.0
go: downloading github.com/mattn/go-colorable v0.1.13
go: downloading golang.org/x/sys v0.29.0
go: downloading github.com/muesli/reflow v0.3.0
go: downloading github.com/sahilm/fuzzy v0.1.1-0.20230530133925-c48e322e2a8f
go: downloading github.com/atotto/clipboard v0.1.4
go: downloading github.com/muesli/termenv v0.15.2
go: downloading github.com/mattn/go-localereader v0.0.1
go: downloading github.com/containerd/console v1.0.4-0.20230313162750-1ae8d489ac81
go: downloading github.com/muesli/ansi v0.0.0-20211018074035-2e021307bc4b
go: downloading github.com/muesli/cancelreader v0.2.2
go: downloading golang.org/x/sync v0.10.0
go: downloading golang.org/x/term v0.28.0
go: downloading github.com/mitchellh/colorstring v0.0.0-20190213212951-d06e56a500db
go: downloading github.com/lucasb-eyer/go-colorful v1.2.0
go: downloading github.com/aymanbagabas/go-osc52/v2 v2.0.1


deletor
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5a009b]

goroutine 1 [running]:
github.com/pashkov256/deletor/internal/tui/views.NewRulesModel({0x656200, 0xc000158370})
	/home/mehanic/.gvm/pkgsets/go1.24.2/global/pkg/mod/github.com/pashkov256/deletor@v1.3.0/internal/tui/views/rules.go:36 +0x4db
github.com/pashkov256/deletor/internal/tui.NewApp({0x656d58, 0x7d5600}, {0x656200, 0xc000158370})
	/home/mehanic/.gvm/pkgsets/go1.24.2/global/pkg/mod/github.com/pashkov256/deletor@v1.3.0/internal/tui/app.go:36 +0x3f
github.com/pashkov256/deletor/internal/tui.Start({0x656d58?, 0x7d5600?}, {0x656200, 0xc000158370})
	/home/mehanic/.gvm/pkgsets/go1.24.2/global/pkg/mod/github.com/pashkov256/deletor@v1.3.0/internal/tui/runner.go:13 +0x27
main.main()
	/home/mehanic/.gvm/pkgsets/go1.24.2/global/pkg/mod/github.com/pashkov256/deletor@v1.3.0/main.go:27 +0x99

--- a/internal/tui/runner.go
+++ b/internal/tui/runner.go
@@ -12,7 +12,6 @@ func Start(
 ) error {
 	app := NewApp(filemanager, rules)
 	p := tea.NewProgram(app, tea.WithAltScreen())
-	rules.SetupRulesConfig()
 	_, err := p.Run()
 	return err
 }

--- a/main.go
+++ b/main.go
@@ -17,10 +17,12 @@ type Task struct {
 }
 
 func main() {
-	config := config.GetFlags()
+	var rules = rules.NewRules()
+
+	rules.SetupRulesConfig()
+	var config = config.GetFlags()
 
 	var fm filemanager.FileManager = filemanager.NewFileManager()
-	rules := rules.NewRules()
 
 	if !config.IsCLIMode {
 		// Start TUI

